tests/PathMappingRepositoryTest.php

```diff
@@ -131,6 +131,79 @@ public function testResolveFilesystemResource()
         $this->assertEquals(Path::normalize(__DIR__.'/Fixtures/dir5/sub'), $resource->getFilesystemPath());
     }
 
+    public function testMergeDirectParent()
+    {
+        $root = __DIR__.'/Fixtures/dir5';
+
+        $this->store->set('/', null);
+        $this->store->set('/webmozart', array($root));
+        $this->store->set('/webmozart/sub', null);
+        $this->store->set('/webmozart/sub/file1', array($root.'/file1'));
+
+        /*
+         * /webmozart/sub should have three children:
+         *      -   file1 from the virtual children
+         *          (directly resolve /webmozart/sub/file1 to /Fixtures/dir5/file1)
+         *      -   file3 and file4 from the real children
+         *          (resolve /webmozart to /Fixtures/dir5 and then list children of /Fixtures/dir5/sub)
+         */
+        $resources = $this->repo->listChildren('/webmozart/sub');
+        $this->assertCount(3, $resources);
```

Comment:
Can you check that the resources are always in the same order? Something like testing `$resources[0]`, `$resources[1]` and `$resources[2]` for their path. That would prevent problems in the future.